examples/end-items/main.js

```diff
@@ -0,0 +1,54 @@
+window.addEventListener('DOMContentLoaded', () => {
+  main();
+});
+
+async function main() {
+  await window.customElements.whenDefined('vertex-viewer');
+  const viewer = document.querySelector('vertex-viewer');
+
+  window.addEventListener('keydown', async (event) => {
+    const scene = await viewer.scene();
+
+    // press 'e' to set end item state
+    if (event.key === 'e') {
+      console.log('setting end item state');
+      await scene
+        .items((op) => op.where((q) => q.withSelected()).setEndItem(true))
+        .execute();
+    }
+
+    // press 'r' to reset end item state
+    if (event.key === 'r') {
+      console.log('resetting end item state');
+      await scene
+        .items((op) => op.where((q) => q.all()).clearEndItem())
+        .execute();
+    }
+  });
+
+  viewer.addEventListener('tap', async (event) => {
+    const { position } = event.detail;
+    const scene = await viewer.scene();
+    const raycaster = await scene.raycaster();
+
+    const result = await raycaster.hitItems(position);
+
+    console.log(result);
+    if (result.hits && result.hits.length > 0) {
+      const hit = result.hits[0];
+
+      await scene
+        .items((op) => [
+          op.where((q) => q.all()).deselect(),
+          op.where((q) => q.withItemId(hit.itemId.hex)).select(),
```

Comment:
We may want to consider leveraging the `ancestors` field on these hit results rather than the `itemId` for the part hit, since it'll always be a leaf item. We may be able to demonstrate that relatively easily by doing something like `hit.ancestors.reverse()[0]` (basically just get the parent assembly), but that might still not always work depending on the assembly structure. What would you think about adding in a quick ancestor selection approach here? I.e. something like:

```typescript
let lastSelectedId;

viewer.addEventListener('tap', async (event) => {
  // setup

  const hit = result.hits[0];
  if (hit.ancestors.length > 0) {
    const ancestors = hit.ancestors.reverse();
    const toSelectId =
      ancestors[ancestors.findIndex((a) => a.itemId.hex === lastSelectedId) + 1]
        .itemId.hex;

    await scene
        .items((op) => [
          op.where((q) => q.all()).deselect(),
          op.where((q) => q.withItemId(toSelectId)).select(),
        ])
        .execute();

    lastSelectedId = toSelectId;
  }
})
```
Thinking that'd give this example a bit more flexibility to select the "correct" assembly to mark as an end item.

Reply:
Good idea! The ancestor selection logic will make setting the end items easier.

What do you think about performing the ancestor selection if the user is holding the shift key? I think there's also value in the user seeing that after the end item is set, a simple selection of any of the leaf items select the entire subassembly.

Reply:
Yeah, good call! I think that makes sense! Seems like the base selection + shift ancestor selection should demonstrate the capability fairly well